### PR TITLE
Variable cards in play

### PR DESCRIPTION
**Pull request by HighPharoah**
- Add a dialog where players can select their difficult level (changes the number of cards in the grid)
- Game will be reset with the new difficulty level

### CODE REVIEW

File: src/card.ts [hunk 1]
```diff
@@ -17,10 +17,11 @@ export interface Card {
...
-const CARD_VALUES = Object.values(SignContent);
+// const CARD_VALUES = Object.values(SignContent);
 
-export function getInitialCards(): Card[] {
-  const date = new Date().toISOString();
+export function getInitialCards(countCardsInPlay: number): Card[] {
+    const date = new Date().toISOString();
+    const CARD_VALUES = Object.values(SignContent).slice(0, countCardsInPlay)
 
   const cards = [...CARD_VALUES, ...CARD_VALUES].map((value, index) => ({
```
Comment (pishposh) [inline src/card.ts]: To randomize which cards get used each game:

```suggestion
  let cardValues = Object.values(SignContent);
  shuffleArray(cardValues);
  cardValues = cardValues.slice(0, countCardsInPlay);

  const cards = [...cardValues, ...cardValues].map((value, index) => ({
```

File: src/components/GameSettings/index.tsx
Comment (mattgrunwald) [inline src/components/GameSettings/index.tsx:1]: Can this use the `Dialog` component I just added?

File: src/game.ts [hunk 2]
```diff
@@ -13,23 +14,49 @@ export interface Game {
...
 
 const DefaultGameData = {
   start: null,
   end: null,
   score: 0,
   attempts: 0,
-  cards: getInitialCards()
-};
+  cards: getInitialCards(getDifficultySpec(Difficulty.EASY).numCards),
+  difficulty: Difficulty.EASY
```
Comment (pishposh) [inline src/game.ts]: @HighPharoah Is this leftover from development? I’d vote for defaulting to more than 2 pairs here.

File: src/card.ts [hunk 3]
```diff
@@ -17,12 +25,14 @@ export interface Card {
   count: number;
 }
 
-const CARD_VALUES = Object.values(SignContent);
-
-export function getInitialCards(): Card[] {
+export function getInitialCards(countCardsInPlay: number): Card[] {
   const date = new Date().toISOString();
+  let cardValues = Object.values(SignContent);
+  //@ts-ignore
```
Comment (pishposh) [inline src/card.ts]:
```suggestion
```

File: src/game.ts (hunk 2 above)
Comment (HighPharoah) [reply src/game.ts]: good catch, updated

**Review by pishposh — APPROVED** [review]
Looks good to me!

**Pull request merged**